src/app.jsx

```diff
@@ -55,38 +59,48 @@ export const Application = () => {
     useEffect(() => {
         if (path === undefined)
             return;
-        setFiles([]);
-        const currentPath = path.slice(0, pathIndex).join("/");
-        const channel = cockpit.channel({
-            payload: "fslist1",
-            path: `/${currentPath}`,
-            superuser: "try",
-            watch: true,
-        });
 
-        channel.addEventListener("message", (ev, data) => {
-            const item = JSON.parse(data);
-            if (item.event === 'present' && item.path[0] !== ".")
-                setFiles(f => [...f, { name: item.path, type: item.type, modified: item.modified }]);
-            else if (item.event === 'deleted') {
-                const deleted_name = item.path.slice(item.path.lastIndexOf("/") + 1);
-                setFiles(f => f.filter((res) => { return res.name !== deleted_name }));
-            } else if (item.event === 'created') {
-                const created_name = item.path.slice(item.path.lastIndexOf("/") + 1);
-                const date_modified = item.tag.slice(item.tag.indexOf("-") + 1, item.tag.lastIndexOf("."));
-                if (created_name[0] !== ".") {
-                    setFiles(f => [...f, { name: created_name, type: item.type, modified: date_modified }]);
+        const getFsList = () => {
+            if (channel.current !== null)
+                channel.current.close();
+
+            const currentPath = path.slice(0, pathIndex).join("/");
+            channel.current = cockpit.channel({
+                payload: "fslist1",
+                path: `/${currentPath}`,
+                superuser: "try",
+                watch: true,
+            });
+
+            const files = [];
+            channel.current.addEventListener("message", (ev, data) => {
+                const item = JSON.parse(data);
+                if (item.event === "present") {
+                    if (item.path[0] !== ".") {
+                        // TODO: Show hidden files if the user has enabled that option
+                        files.push({ ...item, name: item.path });
+                    }
+                } else {
+                    const name = item.path.slice(item.path.lastIndexOf("/") + 1);
+                    if (item.event === 'deleted') {
+                        setFiles(f => f.filter(res => res.name !== name));
+                    } else {
+                        // For events other than 'present' we don't receive file stat information
+                        // so we rerun the fslist command to get the updated information
+                        // https://github.com/allisonkarlitskaya/systemd_ctypes/issues/56
+                        const name = item.path.slice(item.path.lastIndexOf("/") + 1);
+                        if (name[0] !== ".") {
+                            getFsList();
+                        }
```

Comment:
Same here, this assumes that the initial read only has "present" events, otherwise this would be recursive.